lib/extensions/polyglot_with_database/coursemology/polyglot/language.rb

```diff
@@ -0,0 +1,76 @@
+# An abstract language. This is an ActiveRecord class for other models to reference to a family of
+# languages. Languages become concrete and can be referenced in the database when that language
+# implements +Polyglot::Language.instance+.
+#
+# In addition as being an entity which can be referenced in the database, this also adds the
+# scripts and stylesheets needed to syntax highlight code in languages defined here.
+#
+# Each subclass represents a language ancestry, such as differing language versions (see the Python
+# language definition.) Each concrete language has a unique +Polyglot::Language.instance+ which
+# is internally called the +root+ of the language. Derived languages can be defined at runtime to
+# utilise the syntax highlighting capabilities of the root language, while requiring a separate
+# runtime environment to run programs written in the derived language.
+#
+# Do *NOT* remove languages after they have been defined because a database record corresponds to
+# a class implemented here.
+module Extensions::PolyglotWithDatabase::Coursemology::Polyglot::Language
```

Comment:
Yes, the polyglot gem defines only the language definitions, this adds the interaction with ActiveRecord.

I want to cleanly separate them, but I can't find a way to do so :/